File: leetcode/0026.go
package leetcode

func removeDuplicates(nums []int) int {
	n := len(nums)
	if n == 0 {
		return 0
	}
	ans := 0
	for j := ans; j < n; j++ {
		if nums[ans] != nums[j] {
			ans++
			nums[ans] = nums[j]
		}
	}
	return ans + 1
}

File: leetcode/0026.cpp
class Solution {
public:
    int removeDuplicates(vector<int> &nums) {
        int n = nums.size();
        int i = 0;
        for (auto j = 0; j < n; j++) {
            if (nums[i] != nums[j]) {
                nums[++i] = nums[j];
            }
        }
        return n == 0 ? 0 : i + 1;
    }
};